src/TestUtil.hs

```diff
@@ -0,0 +1,93 @@
+module TestUtil where
+
+import           Data.Maybe                      (catMaybes)
+import qualified Data.Text                       as Text
+import           Data.Text                       (Text)
+import qualified Control.Foldl                   as Fold
+import           Control.Exception               (IOException, catch)
+import qualified Turtle                          as Turtle
+import           Turtle                          ((</>), (<.>))
+
+basePath :: Turtle.FilePath
+basePath = "test" </> "resources"
+
+purescriptPath :: Turtle.FilePath
+purescriptPath = basePath </> "purescript"
+
+sourcePath :: Turtle.FilePath
+sourcePath = purescriptPath </> "src"
+
+corefnPath :: Turtle.FilePath
+corefnPath = purescriptPath </> "output"
+
+schemePath :: Turtle.FilePath
+schemePath = basePath </> "scheme"
+
+buildCmd :: Text
+buildCmd = "spago build -u '-g corefn'"
+
+corefnFile :: Text -> Turtle.FilePath
+corefnFile module_ = corefnPath </> Turtle.fromText module_ </> "corefn.json"
+
+schemeFile :: Text -> Turtle.FilePath
+schemeFile module_ = schemePath </> Turtle.fromText module_ <.> "sls"
+
+toText :: Turtle.FilePath -> Text
+toText f = case Turtle.toText f of
+  Left _ -> error "Could not convert FilePath to Text"
+  Right t -> t
+
+moduleName :: Turtle.FilePath -> Text
+moduleName f = Text.intercalate "."
+             $ Text.splitOn "/"
+             $ toText
+             $ Turtle.dropExtension f
+
+findSourceFiles :: IO [Turtle.FilePath]
+findSourceFiles = do
+  paths <- Turtle.fold (Turtle.find (Turtle.ends ".purs") sourcePath) Fold.list
+  let paths' = fmap (Turtle.stripPrefix (sourcePath </> "")) paths
+  return $ catMaybes paths'
+
+findModules :: IO [Text]
+findModules = do
+  files <- findSourceFiles
+  return $ fmap moduleName files
+
+inDir :: Turtle.MonadIO m => Turtle.FilePath -> m b -> m b
+inDir directory do_ = do
+  oldDir <- Turtle.pwd
+  Turtle.cd directory
+  res <- do_
+  Turtle.cd oldDir
+  return res
+
+shellInDir
+  :: Turtle.MonadIO m
+  => Turtle.FilePath
+  -> Text
+  -> m (Turtle.ExitCode, Turtle.Text, Turtle.Text)
+shellInDir dir cmd = inDir dir (Turtle.shellStrictWithErr cmd mempty)
+
+buildCorefn :: IO ()
+buildCorefn = do
+  (exitCode, _out, err) <- shellInDir purescriptPath buildCmd
+  case exitCode of
+    Turtle.ExitSuccess -> pure ()
+    Turtle.ExitFailure _ -> error $ "spago error: " <> show err
+
+removeCorefnFiles :: IO ()
+removeCorefnFiles = catch (Turtle.rmtree corefnPath) handler
+  where
+    handler :: IOException -> IO ()
+    handler _e = pure () -- No such file or directory
+
+removeSchemeFiles :: IO ()
+removeSchemeFiles = do
+  xs <- Turtle.fold (Turtle.find (Turtle.ends ".sls") schemePath) Fold.list
+  mapM_ Turtle.rm xs
```

Comment:
```suggestion
```

As we discussed in the call, let's not remove files here in the executable. This is because it might lead to unexpected deletions - the `Makefile` is the best place to put this if we want such a thing

Reply:
I trust make and shell string/path expansion much less than I trust Haskell. Let's have it properly documented and leave the user do the nasty job instead.